### PR TITLE
Add Match test

### PR DESCRIPTION
**Pull request by mariiatuzovska**

### CODE REVIEW

File: pkg/pair/match_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	encryptedEmails2 := requireEncryptEmails(t, emails[100:], salt, key2)
+	twiceEncryptedEmails1 := requireReEncryptEmails(t, encryptedEmails1, salt, key2)
+	twiceEncryptedEmails2 := requireReEncryptEmails(t, encryptedEmails2, salt, key1)
+	r1, r2 := bytes.NewBuffer(nil), bytes.NewBuffer(nil)
+
+	requireWriteEmails(t, r1, twiceEncryptedEmails1)
+	requireWriteEmails(t, r2, twiceEncryptedEmails2)
+
+	// create map to access encrypted emails faster
+	expectContain := make(map[string]struct{}, 1000)
```
Comment (amanjpro) [inline pkg/pair/match_test.go:34]:
```suggestion
	expectContain := make(map[string]struct{}, 901)
```

It should be exactly 901 emails matched, no?

File: pkg/pair/match_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	require.NoError(t, err, "must stat result file")
+	require.NotZero(t, info.Size(), "must contain data")
+
+	f, err := os.Open(dir + "/result_0.csv")
+	require.NoError(t, err, "must open result file")
+
+	csvReader := csv.NewReader(f)
+	records, err := csvReader.ReadAll()
+	require.NoError(t, err, "must read csv data")
+	require.Len(t, records, 800, "must contain 100 emails")
```
Comment (amanjpro) [inline pkg/pair/match_test.go:59]:
```suggestion
	require.Len(t, records, 800, "must contain 800 emails")
```

File: pkg/pair/match_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+	csvReader := csv.NewReader(f)
+	records, err := csvReader.ReadAll()
+	require.NoError(t, err, "must read csv data")
+	require.Len(t, records, 800, "must contain 800 emails")
+
+	for _, line := range records {
+		require.Len(t, line, 1, "must contain 1 element")
+		_, exists := expectContain[line[0]]
+		require.True(t, exists, "must exist in the hash-encrypted list")
```
Comment (juanli16) [inline pkg/pair/match_test.go:69]:
```suggestion
		_, exists := expectContain[line[0]]
		require.True(t, exists, "must exist in the hash-encrypted list")
		delete(expectContain, line[0])
```

File: pkg/pair/match_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+		csvReader := csv.NewReader(f)
+		records, err := csvReader.ReadAll()
+		require.NoError(t, err, "must read csv data")
+
+		// check if the records exist in the hash-encrypted list
+		for _, line := range records {
+			require.Len(t, line, 1, "must contain 1 element")
+			_, exists := expectContain[line[0]]
+			require.True(t, exists, "must exist in the hash-encrypted list")
```
Comment (juanli16) [inline pkg/pair/match_test.go:133]:
```suggestion
			_, exists := expectContain[line[0]]
			require.True(t, exists, "must exist in the hash-encrypted list")
			delete(expectContain, line[0])
```

File: pkg/pair/match_test.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	"context"
+	"encoding/csv"
+	"fmt"
+	"optable-pair-cli/pkg/io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
```
[the remark this answers is not in the document]
Comment (juanli16) [reply pkg/pair/match_test.go:14]: please use some constants instead

File: pkg/pair/match_test.go (hunk 3 above)
Comment (juanli16) [reply pkg/pair/match_test.go:69]: You don't want the same ID matching over and over again.

**Review by juanli16 — CHANGES_REQUESTED** [review]

**Pull request merged**